### ISSUE TITLE
[Spike] Research Rich Text Editors

### ISSUE THREAD

**Issue by grahamlangford** [pixiebrix/pixiebrix-extension#9289]
## Motivation

* https://www.notion.so/pixiebrix/Image-Upload-and-Rich-Text-Editor-11a43b21a253802198f1c90696f5ec53
* We want to focus on Rich Text Editors and supporting HTML instead of Markdown to allow support for text color, indentation, etc.

## Notes

* Previous spike: https://github.com/pixiebrix/pixiebrix-extension/issues/8978
* Prefer solutions that have a GUI if possible (must support React 17)
* Prefer solutions with MIT/other permissive licenses
* Enterprise/paid solutions may be considered
* We want to be able to handle having images pasted in
    * https://www.notion.so/pixiebrix/Feasibility-Spike-Review-Enabling-image-mod-content-upload-c6711196a72e4847b0569c1746494487
    * https://www.notion.so/pixiebrix/Image-Upload-ERD-12143b21a253805483d9d442d5f3c60e

## Acceptance Criteria

- [x] Create a Notion Doc to record discovery and link to the one above
- [x] Notify Tech Lead and PM for review

## Timebox

* 8 hours

**Comment by mnholtz** [comment]
Findings can be summarized here: https://www.notion.so/pixiebrix/WYSIWG-Research-Spike-12243b21a2538085aa72fac28361da60?pvs=4

Please read the notion for detailed notes on the solutions I came up with.

TL;DR:
- [Quill](https://github.com/slab/quill?tab=readme-ov-file) - if we want to prioritize a quick, plug & chug solution, and aren't so concerned with developer experience in terms of support or extensibility.
- [TipTap](https://github.com/ueberdosis/tiptap?tab=readme-ov-file#examples-codesandbox-and-ui-templates) - if we want to prioritize long-term customization and extensibility, but with a higher initial delivery time as this is a headless library without an out-of-the-box UI. This one is my personal favorite.
- [TinyMCE](https://github.com/tinymce/tinymce) - if we want to prioritize reliable developer experience (e.g. documentation & finding answers to questions online) while still being a quicker plug in chug solution, and less ease in long-term flexibility around custom styles
- [CKEditor5](https://github.com/ckeditor/ckeditor5) - similar to TinyMCE in terms of established documentation, but seems to be more complex with a higher learning curve. This solution might be overkill for what we need, but the flexibility it offers (e.g. it can be optionally headless) is interesting

**Comment by mnholtz** [comment]
Per async convo with @grahamlangford and @Pashaminkovsky I think we are in alignment on TinyMCE
